src/slice_iter.rs

```diff
@@ -1,44 +0,0 @@
-/// Generic Iterator wrapper of slice
```

Comment:
I'll be honest. I have no idea what these slice iterators were for 🤔 maybe wrote them before the API was changed to returning `impl Iterator` but now they can be removed ? a mystery to me now

Reply:
haha luckily you made some good comments in the PRs that informed the changes #94 and #102

Reply:
right but the slice iterators also implement size_hint so I am still at a loss why I did it this way

Reply:
lol 🤷